src/main/webapp/app/exam/manage/student-exams/student-exam-detail.component.ts

```diff
@@ -107,8 +108,29 @@ export class StudentExamDetailComponent implements OnInit {
         const workingTime = this.artemisDurationFromSecondsPipe.transform(this.studentExam.workingTime);
         const workingTimeParts = workingTime.split(':');
         this.workingTimeForm = new FormGroup({
-            minutes: new FormControl(parseInt(workingTimeParts[0] ? workingTimeParts[0] : '0', 10), [Validators.min(0), Validators.required]),
-            seconds: new FormControl(parseInt(workingTimeParts[1] ? workingTimeParts[1] : '0', 10), [Validators.min(0), Validators.max(59), Validators.required]),
+            minutes: new FormControl({ value: parseInt(workingTimeParts[0] ? workingTimeParts[0] : '0', 10), disabled: this.examIsVisible() }, [
+                Validators.min(0),
+                Validators.required,
+            ]),
+            seconds: new FormControl({ value: parseInt(workingTimeParts[1] ? workingTimeParts[1] : '0', 10), disabled: this.examIsVisible() }, [
+                Validators.min(0),
+                Validators.max(59),
+                Validators.required,
+            ]),
         });
     }
+
+    examIsVisible(): boolean {
+        if (this.studentExam.exam) {
+            return moment(this.studentExam.exam.visibleDate).isBefore(moment());
```

Comment:
You may want to add a comment here to make this more clear:
```suggestion
            // Disable the form to edit the working time if the exam is already visible
            return moment(this.studentExam.exam.visibleDate).isBefore(moment());
```